scripts/apps/authoring/authoring/directives/SendItem.js

```diff
@@ -345,6 +402,58 @@ export function SendItem($q, api, desks, notify, authoringWorkspace,
             };
 
             /**
+             * Set the User Action.
+             */
+            scope.setUserAction = function(action) {
+                if (scope.currentUserAction === action) {
+                    return;
+                }
+                scope.currentUserAction = action;
+                storage.setItem(USER_ACTION_KEY, action);
+                setDesksAndStages();
+            };
+
+            /**
+             * Send the current item to different desk or stage and then publish the item
+             */
+            function runSendAndPublish() {
+                var deskId = scope.selectedDesk._id;
+                var stageId = scope.selectedStage._id || scope.selectedDesk.incoming_stage;
+                // send releases lock, increment version.
+                return sendAuthoring(deskId, stageId, scope.selectedMacro, true)
+                    .then(function(item) {
+                        scope.loading = true;
```

Comment:
sendauthoring sets the `scope.loading = false;` and there could be delay in `authoring.open`.